src/string_impls/custom_parsers/hex.rs

```diff
@@ -0,0 +1,177 @@
+// Copyright (C) 2022-2023 Parity Technologies (UK) Ltd. (admin@parity.io)
+// This file is a part of the scale-value crate.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//         http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::{
+    stringify::{ParseError, ParseErrorKind},
+    Value,
+};
+
+/// Attempt to parse a hex string into a [`Value<()>`] (or more specifically,
+/// an unnamed composite).
+///
+/// - Returns an error if we see a leading `0x` and then see invalid hex
+///   characters after this.
+/// - Returns `None` if no `0x` is seen first.
+/// - Returns `Some(value)` if parsing was successful. In this case, the string
+///   reference given is wound forwards to consume what was parsed.
+pub fn parse_hex(s: &mut &str) -> Option<Result<Value<()>, ParseError>> {
+    let bytes = s.as_bytes();
+
+    // Look for leading "0x"; None if this obviously isn't hex.
+    if bytes[0] != b'0' || bytes[1] != b'x' {
```

Comment:
Could the string passed to the function be empty? I think we could be extra safe here and another check to make sure that `bytes[0]` wouldn't panic if the `s.as_bytes()` produces an empty slice

Reply:
Good spot; yup we should def check (and test) for that!